Geometry/EcalCommonData/interface/EcalSimParametersFromDD.h

```diff
@@ -0,0 +1,37 @@
+#ifndef EcalCommonData_EcalSimParametersFromDD_h
+#define EcalCommonData_EcalSimParametersFromDD_h
+
+#include "DetectorDescription/Core/interface/DDsvalues.h"
+#include "DetectorDescription/Core/interface/DDCompactView.h"
+#include "DetectorDescription/DDCMS/interface/DDCompactView.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <string>
+#include <vector>
+
+class DDFilteredView;
+class EcalSimulationParameters;
+
+template <typename T>
```

Comment:
@bsunanda , please do not expose it. One way to do it is to move it to cc:
```
diff --git a/Geometry/EcalCommonData/interface/EcalSimParametersFromDD.h b/Geometry/EcalCommonData/interface/EcalSimParametersFromDD.h
index 80623ac215a..cc7c80e0c02 100644
--- a/Geometry/EcalCommonData/interface/EcalSimParametersFromDD.h
+++ b/Geometry/EcalCommonData/interface/EcalSimParametersFromDD.h
@@ -11,16 +11,6 @@
 class DDFilteredView;
 class EcalSimulationParameters;
 
-template <typename T>
-void myPrint(std::string value, const std::vector<T>& vec) {
-  edm::LogVerbatim("HCalGeom") << "EcalSimParametersFromDD: " << vec.size() << " entries for " << value << ":";
-  unsigned int i(0);
-  for (const auto& e : vec) {
-    edm::LogVerbatim("HCalGeom") << " (" << i << ") " << e;
-    ++i;
-  }
-}
-
 class EcalSimParametersFromDD {
 public:
   EcalSimParametersFromDD() = default;
diff --git a/Geometry/EcalCommonData/src/EcalSimParametersFromDD.cc b/Geometry/EcalCommonData/src/EcalSimParametersFromDD.cc
index c4b8f3e6e83..379ef3d2eee 100644
--- a/Geometry/EcalCommonData/src/EcalSimParametersFromDD.cc
+++ b/Geometry/EcalCommonData/src/EcalSimParametersFromDD.cc
@@ -9,6 +9,17 @@
 #include <iomanip>
 
 //#define EDM_ML_DEBUG
+namespace {
+  template <typename T>
+  void myPrint(std::string value, const std::vector<T>& vec) {
+    edm::LogVerbatim("HCalGeom") << "EcalSimParametersFromDD: " << vec.size() << " entries for " << value << ":";
+    unsigned int i(0);
+    for (const auto& e : vec) {
+      edm::LogVerbatim("HCalGeom") << " (" << i << ") " << e;
+      ++i;
+    }
+  }
+}  // namespace
 
 bool EcalSimParametersFromDD::build(const DDCompactView* cpv, const std::string& name, EcalSimulationParameters& php) {
 #ifdef EDM_ML_DEBUG

```

Reply:
Done